packages/office-ui-fabric-react/src/components/ResizeGroup/examples/ResizeGroup.OverflowSet.Example.tsx

```diff
@@ -41,152 +50,111 @@ function generateData(count: number, cachingEnabled: boolean, checked: boolean):
+
+  const onReduceData = (currentData: any) => {
     if (currentData.primary.length === 0) {
       return undefined;
     }
-
     const overflow = [...currentData.primary.slice(-1), ...currentData.overflow];
     const primary = currentData.primary.slice(0, -1);
-
     let cacheKey = undefined;
-    if (this.state.cachingEnabled) {
+    if (cachingEnabled) {
       cacheKey = computeCacheKey(primary);
     }
     return { primary, overflow, cacheKey };
   };
 
-  private _onGrowData = (currentData: any): any => {
+  const onGrowData = (currentData: any) => {
     if (currentData.overflow.length === 0) {
       return undefined;
     }
-
     const overflow = currentData.overflow.slice(1);
     const primary = [...currentData.primary, ...currentData.overflow.slice(0, 1)];
-
     let cacheKey = undefined;
-    if (this.state.cachingEnabled) {
+    if (cachingEnabled) {
       cacheKey = computeCacheKey(primary);
     }
     return { primary, overflow, cacheKey };
   };
 
-  private _onCachingEnabledChanged = (_: React.FormEvent<HTMLElement | HTMLInputElement>, checked: boolean): void => {
-    this.setState({ cachingEnabled: checked });
-  };
+  const onRenderOverflowButton = (overflowItems: any) => (
+    <CommandBarButton role="menuitem" menuProps={{ items: overflowItems! }} />
+  );
 
-  private _onGrowDataEnabledChanged = (_: React.FormEvent<HTMLElement | HTMLInputElement>, checked: boolean): void => {
-    this.setState({ onGrowDataEnabled: checked });
+  const onRenderData = (data: any) => {
+    return (
+      <OverflowSet
+        role="menubar"
+        items={data.primary}
+        overflowItems={data.overflow.length ? data.overflow : null}
+        onRenderItem={onRenderItem}
+        onRenderOverflowButton={onRenderOverflowButton}
+        styles={overflowSetStyles}
+      />
+    );
   };
 
-  private _onButtonsCheckedChanged = (_: React.FormEvent<HTMLElement | HTMLInputElement>, checked: boolean): void => {
-    this.setState({ buttonsChecked: checked });
+  const onNumberOfItemsChanged = (event: React.FormEvent<HTMLDivElement>, option: IDropdownOption): void => {
+    setNumberOfItems(parseInt(option.text, 10));
   };
 
-  private _onNumberOfItemsChanged = (event: React.FormEvent<HTMLDivElement>, option: IDropdownOption): void => {
-    this.setState({ numberOfItems: parseInt(option.text, 10) });
-  };
-}
+  return (
+    // tslint:disable-next-line:no-constant-condition
+    <div className={false ? styles.resizeIsShort : 'notResized'}>
+      <ResizeGroup
+        role="tabpanel"
+        aria-label="Resize Group with an Overflow Set"
+        data={dataToRender}
+        onReduceData={onReduceData}
+        onGrowData={onGrowDataEnabled ? onGrowData : undefined}
+        onRenderData={onRenderData}
+      />
+      <div className={styles.settingsGroup}>
+        <Checkbox label="Enable caching" onChange={toggleCachingEnabled} checked={cachingEnabled} />
```

Comment:
These should probably all be toggles not checkboxes--toggles are more appropriate when you have on/off settings which take effect instantly. Something to watch for in other examples too.